src/ol/interaction/extentinteraction.js

```diff
@@ -0,0 +1,460 @@
+goog.require('ol.events.Event');
+goog.require('ol.extent');
+goog.require('ol.geom.GeometryLayout');
+goog.require('ol.geom.GeometryType');
+goog.require('ol.geom.Point');
+goog.require('ol.geom.Polygon');
+goog.require('ol.interaction.Pointer');
+goog.require('ol.layer.Vector');
+goog.require('ol.source.Vector');
+
+
+/**
+ * @enum {string}
+ */
+ol.interaction.ExtentEventType = {
+  /**
+   * Triggered after the extent is changed
+   * @event ol.interaction.ExtentEvent
+   * @api
+   */
+  EXTENTCHANGED: 'extentchanged'
+};
+/**
+ * @classdesc
+ * Events emitted by {@link ol.interaction.ExtentInteraction} instances are instances of
+ * this type.
+ *
+ * @constructor
+ * @param {ol.Extent} extent the new extent
+ * @extends {ol.events.Event}
+ */
+ol.interaction.ExtentEvent = function(extent) {
+  goog.base(this, ol.interaction.ExtentEventType.EXTENTCHANGED);
+};
+goog.inherits(ol.interaction.ExtentEvent, ol.events.Event);
+
+/**
+ * @classdesc
+ * Allows the user to draw a vector box by clicking and dragging on the map.
+ * Once drawn, the vector box can be modified by draging its vertices or edges.
+ * This interaction is only supported for mouse devices.
+ *
+ * @constructor
+ * @extends {ol.interaction.Pointer}
+ * @fires ol.ExtentEvent
+ * @param {olx.interaction.ExtentOptions} opts Options.
+ * @api
+ */
+ol.interaction.Extent = function(opts) {
+  /**
+   * Extent Extent
+   * @type {ol.Extent}
+   * @private
+   */
+  this.extent = null;
+  this.pointerHandler = null;
+
+  this.pixelTolerance_ = 10;
+  this.lastPixel_ = null;
+  this.snapToVertex_ = null;
+  this.snappedToVertex_ = null;
+  this.bboxFeature_ = null;
+  this.vertexFeature_ = null;
+  this.bboxOverlay_ = null;
+  this.vertexOverlay = null;
+
+  if (!opts) {
+    opts = {};
+  }
+
+  if (opts.extent) {
+    this.setExtent(opts.extent);
+  }
+
+  /* Inherit ol.interaction.Pointer */
+  goog.base(this, {
+    handleDownEvent: this.handleDownEvent,
+    handleDragEvent: this.handleDragEvent,
+    handleEvent: this.handleEvent,
+    handleUpEvent: this.handleUpEvent
+  });
+
+  /* Geometry Overlays */
+  this.bboxOverlay_ = new ol.layer.Vector({
+    source: new ol.source.Vector({
+      useSpatialIndex: false,
+      wrapX: !!opts.wrapX
+    }),
+    style: opts.boxStyle ? opts.boxStyle : ol.interaction.Extent.getDefaultExtentStyleFunction(),
+    updateWhileAnimating: true,
+    updateWhileInteracting: true
+  });
+  this.vertexOverlay_ = new ol.layer.Vector({
+    source: new ol.source.Vector({
+      useSpatialIndex: false,
+      wrapX: !!opts.wrapX
+    }),
+    style: opts.pointerStyle ? opts.pointerStyle : ol.interaction.Extent.getDefaultPointerStyleFunction(),
+    updateWhileAnimating: true,
+    updateWhileInteracting: true
+  });
+};
+goog.inherits(ol.interaction.Extent, ol.interaction.Pointer);
+
+
+ /* Internal Methods */
+
+/**
+ * @param {ol.MapBrowserEvent} mapBrowserEvent Event.
+ * @return {boolean} Propagate event?
+ * @this {ol.interaction.Extent}
```

Comment:
Ah, scratch that! These handlers are not supposed to be members, but static functions. So `ol.interaction.Extent.handleEvent` instead of `ol.interaction.Extent.prototype.handleEvent` below. Same for the other handlers you pass to the superclass constructor.